airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py

```diff
@@ -620,6 +620,63 @@ def list(self, fields) -> Iterable:
             yield record
 
 
+class CRMAssociationStream(Stream):
+    """  see https://developers.hubspot.com/docs/api/crm/associations for more details
+    """
+
+    entity: Optional[str] = None
+    associations: List[str] = []
+    updated_at_field = None
+    created_at_field = None
+
+    @property
+    def url(self):
+        """Entity URL"""
+        return f"/crm/v3/associations/{self._relationship_from}/{self._relationship_to}/{self._endpoint}"
+
+    def __init__(self, relationship_from: str = None, relationship_to: str = None, endpoint: str = None, **kwargs):
+        super().__init__(**kwargs)
+        self._relationship_from = relationship_from
+        self._relationship_to = relationship_to
+        self._endpoint = endpoint
+
+    def _transform(self, records: Iterable) -> Iterable:
+        """Preprocess record """
+        for record in records:
+            yield record
+
+    def _filter_old_records(self, records: Iterable) -> Iterable:
+        """Skip """
+        for record in records:
+            yield record
+
+
+class DealToContactAssociationsStream(CRMAssociationStream):
```

Comment:
Hello @bazarnov, thanks for the comment and the advices, you are right the `class CRMObjectStream(Stream)` work more than fine, it should be something like this:

    class DealToContactAssociationsStream(CRMObjectStream):
      
      entity = "deal"
      associations = ["contacts"]

Now my question is, how to do a right json schema, since it's almost the same as the deals schema, is it good to duplicate it?

Reply:
You can always go with
```
$ref: schema.json
```
Inside of `your_stream_schema.json`
But if you want you can duplicate it, not a big deal.

Reply:
> Hello @bazarnov, thanks for the comment and the advices, you are right the `class CRMObjectStream(Stream)` work more than fine, it should be something like this:
> 
> 
> 
>     class DealToContactAssociationsStream(CRMObjectStream):
> 
>       
> 
>       entity = "deal"
> 
>       associations = ["contacts"]
> 
> 
> 
> Now my question is, how to do a right json schema, since it's almost the same as the deals schema, is it good to duplicate it?

Sounds great, can wee proceed with reusing those parts, instead?

Reply:
Sure, I'm gonna update the pr with the new changes.